src/client/QXmppClient.cpp

```diff
@@ -320,6 +324,28 @@ bool QXmppClient::isConnected() const
     return d->stream->isConnected();
 }
 
+/// Returns true if the current client state is "active", false if it is
+/// "inactive". See XEP-0352 for details.
+
+bool QXmppClient::isActive() const
+{
+    return d->isActive;
+}
+
+/// Sets the client state as described in to XEP-0352
+
+void QXmppClient::setActive(bool active)
+{
+    if (active != d->isActive && d->stream->isClientStateIndicationEnabled()) {
+        d->isActive = active;
+        QString packet("<%1 xmlns='%2'/>");
+        if (active)
+            d->stream->sendData(packet.arg("active", ns_csi).toUtf8());
```

Comment:
I'd just
```c++
d->stream->sendData(packet.arg(active ? "active" : "inactive", ns_csi).toUtf8());
```
but that might hinder readability a bit, so that's up to you.

Reply:
I like your suggestion, I'll use it. :smiley: